### PR TITLE
Add Fuzz test

### PR DESCRIPTION
**Pull request by jesslatimer**
Adds a fuzz test. After some local testing, this fuzz test seems to have found a couple of bugs/performance issues but nothing critical yet :)

### CODE REVIEW

File: fuzz/fuzz_test.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package fuzz
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/suyashkumar/dicom"
+)
+
+func Fuzz(f *testing.F) {
```
Comment (suyashkumar) [inline fuzz/fuzz_test.go:10]: Can we rename this to specify what's being Fuzzed?
```suggestion
func FuzzDICOMParse(f *testing.F) {
```

File: fuzz/fuzz_test.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package fuzz
+
```
Comment (suyashkumar) [inline fuzz/fuzz_test.go:2]: Can we rename this file to be dicom_fuzz_test.go?

**Review by suyashkumar — CHANGES_REQUESTED** [review]
Thank you for the contribution, it is a great idea to add some Fuzzing! Just had a couple minor comments before merging.

File: fuzz/fuzz_test.go (hunk 1 above)
Comment (jesslatimer) [reply fuzz/fuzz_test.go]: Done

File: fuzz/fuzz_test.go (hunk 2 above)
Comment (jesslatimer) [reply fuzz/fuzz_test.go]: Done

**Pull request merged**